docs/examples/PF_OOSM_example.py

```diff
@@ -0,0 +1,325 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+====================================================
+Particle filtering with Out-of-sequence Measurements
+====================================================
+"""
+
+# %%
+# Out-of-sequence measurements (OOSM) are a common and bothersome problem in tracking,
+# arising from delays in the processing chain and frequent when dealing with
+# multi-sensor scenarios.
+#
+# In literature, there are a number of approaches to deal with this problem (see [#]_ and [#]_)
+# and in a series of examples we have presented some algorithms that deal with such measurements.
+# These examples were adopted using a Kalman filter approach, here instead we consider the
+# case of Particle filters (the algorithm implementation can be found here [#]_).
+#
+# Particle filters (PF) work in a different manner compared to the Kalman filters because they sequentially
+# update the distribution, while the latter updates only the filtered distribution.
+# In the PF application, we have each trajectory defined as a particle, where each have an associated weight.
+# Weights can vary as new information arrives (measurements or clutter).
+#
+# The algorithm presented follows this logic: we have a series of measurements that arrive to the detector at
+# different consecutive timesteps, :math:`k=0,1,2...`, and their detection time is defined as :math:`t_{k}`.
+# If :math:`t_{k} > t_{k-1}`, meaning :math:`t_{k}` is consecutive to :math:`t_{k-1}`, we apply the stadard
+# particle filter tracking method.
+# Instead, if :math:`t_{k} < t_{k-1}`, a delayed measurement, then we apply this algorithm.
+# We look in the list of measurements where :math:`t_{k}` belongs.
+# In detail, we look for two indices, :math:`a` and :math:`b`, such that :math:`t_{a} > t_{k} > t_{b}`.
+# In this manner, we are able to insert the measurement in each particle tracjectory history at the right time.
```

Comment:
```suggestion
# We have a series of measurements that arrive to the detector at
# different consecutive timesteps, :math:`k=0,1,2...`, with detection time defined as :math:`t_{k}`.
# If :math:`t_{k} > t_{k-1}`, meaning :math:`t_{k}` is consecutive to :math:`t_{k-1}`, we apply the standard
# particle filter tracking method.
# Instead, if :math:`t_{k} < t_{k-1}`, a delayed measurement, then we apply the following algorithm:
# look in the list of measurements where :math:`t_{k}` belongs and find two indices, :math:`a` and :math:`b`, such that :math:`t_{a} > t_{k} > t_{b}`.
# In this manner, we are able to insert the measurement in each particle trajectory history at the right time.
```

Reply:
Suggested changes to improve grammar and readability. Check I've not changed the meaning of the text - feel free to ignore/change what I've suggested if you disagree.